pages/_app.tsx

```diff
@@ -45,14 +57,46 @@ const App = ({
     process.env.NEXT_PUBLIC_AMPLITUDE_API_KEY ??
       '5416da0efc30dc892889733916be497b'
   )
+
+  const network = useMemo(() => {
+    switch (cluster) {
+      case 'mainnet':
+        return WalletAdapterNetwork.Mainnet
+      case 'devnet':
+        return WalletAdapterNetwork.Devnet
+      case 'testnet':
+        return WalletAdapterNetwork.Testnet
+      default:
+        return WalletAdapterNetwork.Mainnet
+    }
+  }, [cluster])
+
+  const wallets = useMemo(
+    () => [
+      new PhantomWalletAdapter(),
+      new BackpackWalletAdapter(),
+      new SolflareWalletAdapter({ network }),
+      new CoinbaseWalletAdapter(),
+      new BraveWalletAdapter(),
+      new FractalWalletAdapter(),
+      new GlowWalletAdapter({ network }),
+      new LedgerWalletAdapter(),
+      new TorusWalletAdapter({ params: { network, showTorusButton: false } }),
+    ],
+    [network]
+  )
```

Comment:
Instantiate the adapters only when the network changes.

Reply:
going to add slope as well :) some people still use it so weve heard when it got removed for a bit

Reply:
I pity the fool